src/services/services.ts

```diff
@@ -1887,23 +1887,21 @@ namespace ts {
             synchronizeHostData();
             const sourceFile = getValidSourceFile(fileName);
             const span = createTextSpanFromBounds(start, end);
-            const newLineCharacter = getNewLineOrDefaultFromHost(host);
             const formatContext = formatting.getFormatContext(formatOptions);
 
             return flatMap(deduplicate(errorCodes, equateValues, compareValues), errorCode => {
                 cancellationToken.throwIfCancellationRequested();
-                return codefix.getFixes({ errorCode, sourceFile, span, program, newLineCharacter, host, cancellationToken, formatContext });
+                return codefix.getFixes({ errorCode, sourceFile, span, program, host, cancellationToken, formatContext });
             });
         }
 
         function getCombinedCodeFix(scope: CombinedCodeFixScope, fixId: {}, formatOptions: FormatCodeSettings): CombinedCodeActions {
             synchronizeHostData();
             Debug.assert(scope.type === "file");
             const sourceFile = getValidSourceFile(scope.fileName);
-            const newLineCharacter = getNewLineOrDefaultFromHost(host);

```

Comment:
we should mark this function as deprecated, and replace all its uses as well.

Reply:
It appears to be internal.  Can we just delete it?

Reply:
Actually, the remaining callers look pretty legitimate.  Is there another method they should call instead?

Reply:
removign it should be fine then.

Reply:
I'm confused.  This method appears to have callers.  Why are we deprecating/removing it?

Reply:
looking at this again, seems like in every call to this method, we have an insufficient public API. for instance, in `getDocCommentTemplateAtPosition`, seems like the public API should take a `FormatCodeSettings` just like `getCodeFixesAtPosition` does. 
I do not think this is related to this PR per se, but this is something that we should fix for the long term. unfortunately we can not remove the `getNewLineOrDefaultFromHost` as you noted without breaking our public API, but we should add optional `FormatCodeSettings` in places where we use it to allow the host to override the value.

Reply:
@mhegazy Can you please confirm that the latest commit (https://github.com/Microsoft/TypeScript/pull/21158/commits/d97dec85749ce886fcafb9d93082525206b40ec1) is what you had in mind (not counting a bug about reviewing callers of `getNewLineOrDefaultFromHost`)?

Reply:
👍

Reply:
Thanks!  Filed https://github.com/Microsoft/TypeScript/issues/21289.